main/parse.c

```diff
@@ -54,6 +55,8 @@ typedef struct {
 	enum specType specType;
 }  parserCandidate;
 
+static ptrArray *parsersUsedInCurrentInput;
```

Comment:
Hum, actually it's probably better if the array is created and destroyed explicitly, isn't it?  Right now it would be found as a `still reachable` leak in Valgrind, and IIRC you tried to get rid of those in the parsers, so I guess the same applies to the main part, doesn't it?